dimod/reference/samplers/exact_solver.py

```diff
@@ -20,14 +20,18 @@
     energy for every possible sample, they are very slow.
 """
 import numpy as np
+from itertools import product
 
 from dimod.core.sampler import Sampler
-from dimod.sampleset import SampleSet, as_samples
+from dimod.sampleset import SampleSet
 from dimod.core.polysampler import PolySampler
 from dimod.vartypes import Vartype
-from dimod.exceptions import SamplerUnknownArgWarning
 
-__all__ = ['ExactSolver', 'ExactPolySolver', 'ExactDQMSolver']
+from typing import TYPE_CHECKING
```

Comment:
Not to be terribly pedantic, but let's keep our imports clean by following [Pep8](https://www.python.org/dev/peps/pep-0008/#imports). Specifically
1) Standard library imports.
2) Related third party imports.
3) Local application/library specific imports.

so should be
```python
from itertools import product
from typing import TYPE_CHECKING

import numpy as np

from dimod.core.sampler import ...